### DISCUSSION TITLE
identify: Unable to decompress TIF image

### DISCUSSION THREAD

**Discussion by DavidSrwld** [ImageMagick/ImageMagick#3309] [General]
**Prerequisites**
 I have written a descriptive issue title
 I have verified that I am using the latest version of ImageMagick

**Description**
Got error on tif parsing (convert, identify).

**Steps to Reproduce:**
Execute magick identify -verbose C:\...\incorrect.tif
Get errors
```
identify: Incompatible type for "RichTIFFIPTC"; tag ignored. `TIFFFetchNormalTag' @ warning/tiff.c/TIFFWarnings/960.
identify: unable to decompress image `C:\...\incorrect.tif' @ error/psd.c/ReadPSDChannel/1539.
```
Execute magick identify -verbose C:\...\incorrect.tif[0]
No errors, correct validation. Also correct conversion to .jpg, but with a different color density.

Additional Information:
Saving the TIF in PS2019 and later will lead to this error. Re-save with same options in PS2018 will not cause this problem.
![image-2021-02-10-16-03-00-200](https://user-images.githubusercontent.com/69974988/109307313-b7185f80-7840-11eb-9296-cacf0341f1ad.png)

**ImageMagick version:**
Version: ImageMagick 7.0.11-0 Q16 x64 2021-13-02
**Environment (Operating system, version and so on):**
Windows 10 x64

**Comment by snibgo** [comment]
Please supply a link to incorrect.tif

**Reply by DavidSrwld** [reply]
The TIFs are used for commercial reasons and can't be publicated... But maybe it is a common/known problem

**Comment by fmw42** [comment]
Post some other TIFF that is not proprietary.  You can zip them and that can be uploaded.  I note that IM uses libtiff, which cannot deal with the old style JPG compression. That might be your issue. Check if it is JPG compressed.

**Reply by DavidSrwld** [reply]
Compression is LZW, in both cases (successfull and not successfull)

**Comment by snibgo** [comment]
Please paste here the text output from these two commands:
```
magick incorrect.tif -verbose info:

exiftool incorrect.tif
```

**Reply by DavidSrwld** [reply]
Image:
  Filename: incorrect.tif
  Format: TIFF (Tagged Image File Format)
  Mime type: image/tiff
  Class: DirectClass
  Geometry: 733x598+0+0
  Resolution: 300x300
  Print size: 2.44333x1.99333
  Units: PixelsPerInch
  Colorspace: CMYK
  Type: ColorSeparation
  Endianness: LSB
  Depth: 8-bit
  Channel depth:
    Cyan: 8-bit
    Magenta: 1-bit
    Yellow: 1-bit
    Black: 8-bit
  Channel statistics:
    Pixels: 438334
    Cyan:
      min: 0  (0)
      max: 255 (1)
      mean: 24.9353 (0.0977856)
      median: 0 (0)
      standard deviation: 72.5987 (0.284701)
      kurtosis: 5.62485
      skewness: 2.72804
      entropy: 0.123138
    Magenta:
      min: 0  (0)
      max: 0 (0)
      mean: 0 (0)
      median: 0 (0)
      standard deviation: 0 (0)
      kurtosis: -3
      skewness: 0
      entropy: 0
    Yellow:
      min: 0  (0)
      max: 0 (0)
      mean: 0 (0)
      median: 0 (0)
      standard deviation: 0 (0)
      kurtosis: -3
      skewness: 0
      entropy: 0
    Black:
      min: 0  (0)
      max: 255 (1)
      mean: 20.6596 (0.081018)
      median: 0 (0)
      standard deviation: 51.8477 (0.203324)
      kurtosis: 8.23978
      skewness: 2.86822
      entropy: 0.204292
  Image statistics:
    Overall:
      min: 0  (0)
      max: 255 (1)
      mean: 11.3987 (0.0447009)
      median: 0 (0)
      standard deviation: 31.1116 (0.122006)
      kurtosis: 19.1763
      skewness: 4.43161
      entropy: 0.0818576
  Total ink density: 140%
  Rendering intent: Perceptual
  Gamma: 0.454545
  Chromaticity:
    red primary: (0.64,0.33)
    green primary: (0.3,0.6)
    blue primary: (0.15,0.06)
    white point: (0.3127,0.329)
  Matte color: grey74
  Background color: white
  Border color: srgb(223,223,223)
  Transparent color: none
  Interlace: None
  Intensity: Undefined
  Compose: Over
  Page geometry: 733x598+0+0
  Dispose: Undefined
  Iterations: 0
  Compression: LZW
  Orientation: TopLeft
  Profiles:
    Profile-8bim: 10520 bytes
    Profile-tiff:37724: 44480892 bytes
    Profile-xmp: 14746 bytes
  Properties:
    date:create: 2021-02-19T10:35:20+00:00
    date:modify: 2021-02-11T13:09:38+00:00
    dc:format: image/tiff
    exif:ColorSpace: 65535
    exif:PixelXDimension: 733
    exif:PixelYDimension: 598
    photoshop:ColorMode: 4
    signature: 7852967c365df5ea95ff121eccc64f994c9848a327d72e7da807cd5992fb2bee
    tiff:alpha: unspecified
    tiff:endian: lsb
    tiff:photometric: separated
    tiff:rows-per-strip: 89
    tiff:software: Adobe Photoshop CC 2019 (Windows)
    tiff:timestamp: 2020:09:11 15:09:48
    xmp:CreateDate: 2020-01-14T11:47:59+01:00
    xmp:CreatorTool: Adobe Photoshop 21.2 (Windows)
    xmp:MetadataDate: 2020-09-11T15:09:48+02:00
    xmp:ModifyDate: 2020-09-11T15:09:48+02:00
    xmpMM:DocumentID: adobe:docid:photoshop:bed8e0e2-4a0c-f747-b3c9-94f2c5367e6c
    xmpMM:InstanceID: xmp.iid:c749d3b6-e71d-ec42-8c17-57886482a9b8
    xmpMM:OriginalDocumentID: xmp.did:30acdf1e-eb36-1445-ac4e-d5df8712a6aa
  Artifacts:
    verbose: true
  Tainted: False
  Filesize: 42.5596MiB
  Number pixels: 438334
  Pixels per second: 778456P
  User time: 0.563u
  Elapsed time: 0:01.563
  Version: ImageMagick 7.0.11-0 Q16 x64 2021-02-13 https://imagemagick.org
identify: Incompatible type for "RichTIFFIPTC"; tag ignored. `TIFFFetchNormalTag' @ warning/tiff.c/TIFFWarnings/960.
identify: unable to decompress image `incorrect.tif' @ error/psd.c/ReadPSDChannel/1539.

**Comment by snibgo** [comment]
You didn't show the text output from:
```
exiftool incorrect.tif
```
The metadata from "magick incorrect.tif -verbose info:" looks okay. Perhaps "-debug all" may give a clue to why IM can't decompress the image. Perhaps Photoshop has used a compression algorithm that libtiff hasn't (yet) implemented.
